src/modules/featured/featured.service.ts

```diff
@@ -94,7 +86,38 @@ export class FeaturedService {
     }
   }
 
-  private getFeaturedCollectionsCacheKey() {
-    return generateCacheKeyFromParams('featuredCollections');
+  async addFeaturedCollection(
+    collection: string,
+    type: FeaturedCollectionTypeEnum,
+  ): Promise<boolean> {
+    const isAdded = await this.persistenceService.addFeaturedCollection(
+      collection,
+      type,
+    );
+    if (isAdded) {
+      await this.invalidateFeaturedCollectionsCache();
+    }
+    return isAdded;
+  }
+
+  async removeFeaturedCollection(collection: string): Promise<boolean> {
+    const isRemoved = await this.persistenceService.removeFeaturedCollection(
+      collection,
+    );
+    if (isRemoved) {
+      await this.invalidateFeaturedCollectionsCache();
+    }
+    return isRemoved;
+  }
+
+  async invalidateFeaturedCollectionsCache(): Promise<void> {
+    await Promise.all([
```

Comment:
this in not needed here, and maybe rename this method trigger cache invalidation and keep only the trigger event.

Reply:
Done